ta4j-core/src/main/java/org/ta4j/core/num/DoubleNum.java

```diff
@@ -247,12 +297,31 @@ public boolean isLessThanOrEqual(Num other) {
 
     @Override
     public Num min(Num other) {
-        return other.isNaN() ? NaN : new DoubleNum(Math.min(delegate, ((DoubleNum) other).delegate));
+        double a = delegate;
+        // noinspection ConstantConditions
+        if (a != a)
+            return NaN; // a is NaN
+        double b = other.getDelegate().doubleValue();
+        if ((a == 0d) && (b == 0d) && (Double.doubleToRawLongBits(b) == negativeZeroDoubleBits)) {
+            // Raw conversion ok since NaN can't map to -0d.
+            return other;
+        }
+        return (a <= b) ? this : other;
+
     }
 
     @Override
     public Num max(Num other) {
-        return other.isNaN() ? NaN : new DoubleNum(Math.max(delegate, ((DoubleNum) other).delegate));
+        double a = delegate;
+        // noinspection ConstantConditions
```

Comment:
very cryptic, use a maintainable and user friendly description.

Reply:
@jnorthrup I don't care which IDE you use and ta4j-code does not care it also. So please don't pollute ta4j-code with your IDE hints or try to use an appropriate coding style where the IDE doesn't complain.